src/java.base/share/classes/javax/crypto/KDF.java

```diff
@@ -0,0 +1,443 @@
+        try {
+            instance = getInstance(algorithm, (AlgorithmParameterSpec) null, provider);
+        } catch (InvalidParameterSpecException e) {
+            throw new NoSuchAlgorithmException(
+                "Received an InvalidParameterSpecException. Does this algorithm require an "
+                + "AlgorithmParameterSpec?");
+        }
+        return instance;
+    }
+
+    /**
+     * Creates an instance of the {@code KDF} object using a supplied {@code Provider} instance.
+     *
+     * @param algorithm
+     *     the key derivation algorithm to use
+     * @param provider
+     *     the provider
+     *
+     * @return a {@code KDF} object
+     *
+     * @throws NoSuchAlgorithmException
+     *     if no {@code Provider} supports a {@code KDFSpi} implementation for the specified
+     *     algorithm.
+     */
+    public static KDF getInstance(String algorithm, Provider provider)
+        throws NoSuchAlgorithmException {
+        KDF instance = null;
+        try {
+            instance = getInstance(algorithm, (AlgorithmParameterSpec) null, provider);
+        } catch (InvalidParameterSpecException e) {
+            throw new NoSuchAlgorithmException(
+                "Received an InvalidParameterSpecException. Does this algorithm require an "
+                + "AlgorithmParameterSpec?");
+        }
+        return instance;
+    }
+
+    /**
+     * Creates an instance of the {@code KDF} object.
+     *
+     * @param algorithm
+     *     the key derivation algorithm to use
+     * @param algParameterSpec
+     *     the {@code AlgorithmParameterSpec} used to configure this KDF's algorithm or {@code null}
+     *     if no additional parameters were provided.
+     *
+     * @return a {@code KDF} object
+     *
+     * @throws NoSuchAlgorithmException
+     *     if no {@code Provider} supports a {@code KDFSpi} implementation for the specified
+     *     algorithm.
+     * @throws InvalidParameterSpecException
+     *     if the {@code AlgorithmParameterSpec} is an invalid value
+     */
+    public static KDF getInstance(String algorithm, AlgorithmParameterSpec algParameterSpec)
+        throws NoSuchAlgorithmException, InvalidParameterSpecException {
+        Objects.requireNonNull(algorithm, "null algorithm name");
+        try {
+            Instance instance = GetInstance.getInstance("KDF", KDFSpi.class, algorithm,
+                                                        algParameterSpec);
+            if (!JceSecurity.canUseProvider(instance.provider)) {
+                String msg = "JCE cannot authenticate the provider " + instance.provider.getName();
+                throw new SecurityException(msg);
+            }
+            return new KDF((KDFSpi) instance.impl, instance.provider, algorithm, algParameterSpec);
+
+        } catch (NoSuchAlgorithmException nsae) {
+            throw new NoSuchAlgorithmException("Algorithm " + algorithm + " not available", nsae);
+        }
```

Comment:
Why do you catch and rethrow the NSAE exception? Also, `GetInstance.getInstance` does not throw `InvalidParameterSpecException`. You will need to catch the NSAE and see if the cause is an IPSE and then rethrow as an IPSE - see the `CertStore` code for an example.